src/Core/src/Platform/iOS/WrapperView.cs

```diff
@@ -122,22 +142,34 @@ public override void SetNeedsLayout()
 		void SetClip()
 		{
 			var mask = MaskLayer;
+			var backgroundMask = BackgroundMaskLayer;
 
-			if (mask == null && Clip == null)
+			if (mask is null && Clip is null)
 				return;
 
-			mask ??= MaskLayer = new CAShapeLayer();
 			var frame = Frame;
 			var bounds = new RectF(0, 0, (float)frame.Width, (float)frame.Height);
-
 			var path = _clip?.PathForBounds(bounds);
 			var nativePath = path?.AsCGPath();
+
+			mask ??= MaskLayer = new CAShapeLayer();
 			mask.Path = nativePath;
+
+			var backgroundLayer = GetBackgroundLayer();
+
+			// We wrap some controls for certain visual effects like applying background gradient etc.
+			// For this reason, we have to clip the background layer as well if it exists.
+			if (backgroundLayer is null)
```

Comment:
This is the main change.

Reply:
We wrap some controls for certain visual effects like applying background gradient etc.
For this reason, we have to clip the background layer as well if it exists.